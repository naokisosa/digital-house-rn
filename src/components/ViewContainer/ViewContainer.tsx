import { StyleSheet, ViewStyle } from 'react-native';
import React from 'react';
import { Edge, SafeAreaView } from 'react-native-safe-area-context';
import { colors } from '@style';

interface Props {
  children?: React.ReactNode;
  edges?: Edge[];
  style?: ViewStyle;
  header?: React.ReactNode;
}

const DEFAULT_EDGES: Edge[] = ['bottom', 'left', 'right', 'top'];
const HEADER_EDGES: Edge[] = ['bottom', 'left', 'right'];

const ViewContainer = ({ children, edges = DEFAULT_EDGES, style, header }: Props) => {
  return (
    <>
      {header}
      <SafeAreaView style={[styles.container, style]} edges={header ? HEADER_EDGES : edges}>
        {children}
      </SafeAreaView>
    </>
  );
};

export default ViewContainer;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.backgroundColor,
    paddingHorizontal: 20,
  },
});
