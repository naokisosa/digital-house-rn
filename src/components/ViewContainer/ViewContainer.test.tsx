import 'react-native';
import React from 'react';

import ViewContainer from './ViewContainer';
import renderer from 'react-test-renderer';
import { Text } from 'react-native';

describe('Test ViewContainer component', () => {
  it('renders correctly with children', () => {
    const { toJSON } = renderer.create(
      <ViewContainer>
        <Text>Mock</Text>
      </ViewContainer>
    );
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly with children and header', () => {
    const { toJSON } = renderer.create(
      <ViewContainer header={<Text>Header</Text>}>
        <Text>Mock</Text>
      </ViewContainer>
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
