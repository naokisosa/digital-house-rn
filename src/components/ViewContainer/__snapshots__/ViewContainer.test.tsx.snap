// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ViewContainer component renders correctly with children 1`] = `
<RNCSafeAreaView
  edges={
    [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    [
      {
        "backgroundColor": "#F8F8F8",
        "flex": 1,
        "paddingBottom": 6,
        "paddingHorizontal": 20,
        "paddingTop": 11,
      },
      undefined,
    ]
  }
>
  <Text>
    Mock
  </Text>
</RNCSafeAreaView>
`;

exports[`Test ViewContainer component renders correctly with children and header 1`] = `
[
  <Text>
    Header
  </Text>,
  <RNCSafeAreaView
    edges={
      [
        "bottom",
        "left",
        "right",
      ]
    }
    style={
      [
        {
          "backgroundColor": "#F8F8F8",
          "flex": 1,
          "paddingBottom": 6,
          "paddingHorizontal": 20,
          "paddingTop": 11,
        },
        undefined,
      ]
    }
  >
    <Text>
      Mock
    </Text>
  </RNCSafeAreaView>,
]
`;
